homeassistant/components/sensor/time_date.py

```diff
@@ -56,7 +59,7 @@ def __init__(self, option_type):
         self._name = OPTION_TYPES[option_type]
         self.type = option_type
         self._state = None
-        self.update()
+        self.async_update()
```

Comment:
I'll handle that with async_add_device callback and force_update parameter. I make a PR, after that you can use this on this place :+1: